erpnext/controllers/selling_controller.py

```diff
@@ -404,7 +404,7 @@ def update_reserved_qty(self):
 			if so and so_item_rows:
 				sales_order = frappe.get_doc("Sales Order", so)
 
-				if sales_order.status in ["Closed", "Cancelled"]:
+				if (sales_order.status in ["Closed", "Cancelled"] and not self.is_return) or (sales_order.status in ["Cancelled"] and self.is_return):
```

Comment:
Should not be allowed for cancelled orders
```suggestion
				if (sales_order.status == "Closed" and not self.is_return) or sales_order.status in ["Cancelled"]:
```